querybook/server/logic/data_element.py

```diff
@@ -112,25 +127,20 @@ def create_data_element_association(
     primitive_type: str = None,
     session=None,
 ):
-    if (
-        not data_element_tuple
-        or not data_element_tuple.name
-        or not data_element_tuple.type
-    ):
-        return None
-
-    if type(data_element_tuple) is str:
-        data_element = get_data_element_by_name(data_element_tuple, session=session)
-    else:
-        data_element = create_or_update_data_element(
-            metastore_id, data_element_tuple, session=session
-        )
-
-    if not data_element and not primitive_type:
+    if not data_element_tuple and not primitive_type:
         raise Exception(
             f"Can not create DataElementAssociation: {data_element_tuple} is not a valid data element and primitive type is empty"
         )
 
+    data_element = None
+    if data_element_tuple:
+        if type(data_element_tuple) is str:
+            data_element = get_data_element_by_name(data_element_tuple, session=session)
+        elif data_element_tuple.name and data_element_tuple.type:
+            data_element = create_or_update_data_element(
+                metastore_id, data_element_tuple, session=session
+            )
+
```

Comment:
does it need to check the `data_element is None` again here?

Reply:
i added a check on line 148